main/main.go

```diff
@@ -39,159 +45,204 @@ const (
 	seqNumberSetError            // 6
 	statusReportingError         // 7
 	settingsNotFoundError        // 8
+	versionMismatchError         // 9
+	jsonParsingError             // 10
 )
 
-func install() {
-	infoLogger.Println("Installed Succesfully.")
-}
-
-func enable() {
-	infoLogger.Println("Enabled Successfully.")
-}
-
-func disable() {
-	infoLogger.Println("Disabled Successfully.")
+// extension specific PublicSettings
+type extensionPublicSettings struct {
+	Name string `json:"name"`
 }
 
-func uninstall() {
-	infoLogger.Println("Uninstalled Successfully.")
+// extension specific ProtectedSettings
+type extensionPrivateSettings struct {
+	SecretString       string   `json:"secretString"`
+	SecretScript       string   `json:"secretScript"`
+	FileURLs           []string `json:"fileUris"`
+	StorageAccountName string   `json:"storageAccountName"`
+	StorageAccountKey  string   `json:"storageAccountKey"`
 }
 
-func update() {
-	infoLogger.Println("Updated Successfully.")
-}
+func install() {
+	operation := "install"
+	infoLogger.Printf("Extension MrSeq: %d, Environment MrSeq: %d", extensionMrSeq, environmentMrSeq)
+	operationLogger.Printf("[Seq Num: %d] [operation: %s]", environmentMrSeq, operation)
 
-func parseJSON(filename string) error {
-	//	Open the provided file
-	jsonFile, err := os.Open(filename)
+	err := status.ReportTransitioning(environmentMrSeq, operation, "installation in progress")
+	infoLogger.Println("Installation in progress")
 	if err != nil {
-		return errors.Wrapf(err, "Failed to open \"%s\"", filename)
+		errorLogger.Printf("Status reporting error: %+v", err)
+		os.Exit(statusReportingError)
 	}
-	infoLogger.Println("File opened successfully")
 
-	// Defer file closing until parseJSON() returns
-	defer jsonFile.Close()
-
-	//	Unmarshall the bytes from the JSON file
-	// 	TODO: If we know the exact format, we can read the JSON into a struct which might be cleaner
-	byteValue, _ := ioutil.ReadAll(jsonFile)
-	var result map[string]interface{}
-	json.Unmarshal([]byte(byteValue), &result)
-
-	// Get the map with the name "keys"
-	keys := result["keys"].(map[string]interface{})
-
-	//	Parse each key value and reverse the string by appending characters backwards
-	for key, value := range keys {
-		reverseValue := ""
-		for _, val := range value.(string) {
-			reverseValue = string(val) + reverseValue
-		}
-
-		infoLogger.Println(key, reverseValue)
+	err = status.ReportSuccess(environmentMrSeq, operation, "installation is complete")
+	if err != nil {
+		errorLogger.Printf("Status reporting error: %+v", err)
+		os.Exit(statusReportingError)
 	}
-	return nil
+	infoLogger.Println("Installation is complete")
+	os.Exit(successfulExecution)
```

Comment:
Similarly, dont exit separately from each function even on success, only exit from the main() function so that its common for all functions